Clang 15 Build failed after #49752

Using Clang 15 build latest code with #49752 will failed:
```

[2023-06-13T23:38:02.611Z] /ClickHouse/src/Formats/CapnProtoSerializer.cpp:754:9: error: missing 'typename' prior to dependent type name 'CapnpType::Reader'
[2023-06-13T23:38:02.611Z]         CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
[2023-06-13T23:38:02.611Z]         ^~~~~~~~~~~~~~~~~
[2023-06-13T23:38:02.611Z]         typename 
[2023-06-13T23:38:02.611Z] /ClickHouse/src/Formats/CapnProtoSerializer.cpp:804:9: error: missing 'typename' prior to dependent type name 'CapnpType::Reader'
[2023-06-13T23:38:02.611Z]         CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
[2023-06-13T23:38:02.611Z]         ^~~~~~~~~~~~~~~~~
[2023-06-13T23:38:02.611Z]         typename 
```

Clang 16 is ok

--- a/src/Formats/CapnProtoSerializer.cpp
+++ b/src/Formats/CapnProtoSerializer.cpp
@@ -751,7 +751,7 @@ namespace
     private:
         using Reader = typename CapnpType::Reader;
 
-        CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
+        typename CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
         {
             auto data = column->getDataAt(row_num);
             if constexpr (std::is_same_v<CapnpType, capnp::Data>)
@@ -801,7 +801,7 @@ namespace
     private:
         using Reader = typename CapnpType::Reader;
 
-        CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
+        typename CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
         {
             auto data = column->getDataAt(row_num);
             if constexpr (std::is_same_v<CapnpType, capnp::Data>)

--- a/src/Formats/CapnProtoSerializer.cpp
+++ b/src/Formats/CapnProtoSerializer.cpp
@@ -751,7 +751,7 @@ namespace
     private:
         using Reader = typename CapnpType::Reader;
 
-        typename CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
+        Reader getData(const ColumnPtr & column, size_t row_num)
         {
             auto data = column->getDataAt(row_num);
             if constexpr (std::is_same_v<CapnpType, capnp::Data>)
@@ -801,7 +801,7 @@ namespace
     private:
         using Reader = typename CapnpType::Reader;
 
-        typename CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
+        Reader getData(const ColumnPtr & column, size_t row_num)
         {
             auto data = column->getDataAt(row_num);
             if constexpr (std::is_same_v<CapnpType, capnp::Data>)

